### ISSUE TITLE
Network map is incorrect when interfaces share MACs

### ISSUE THREAD

**Issue by keeperofdakeys** [librenms/librenms#3573]
I've got an openwrt router that does inter-vlan routing, and the wrong interfaces are showing on the network map (MAC mode). This seems to be occurring because the router has multiple interfaces with the same MAC address (eg. eth0, eth0.100, eth0.110, etc...), and the SQL statement used for the network map only considers the mac address of the remote port.

Since librenms usually has the ip addresses of ports, the network map code could also consider this to create the local <-> remote port associations. While it's nowhere near robust enough (doesn't consider ports with no ip, or multiple ips), adding these sql statements to html/includes/print-map.inc.php fixes the problem for me.

Join Clause:
``` LEFT JOIN `ipv4_addresses` AS `P2_ip` ON `P2_ip`.`port_id` = `P2`.`port_id````

Where Clause:
``` `P2_ip`.`ipv4_address` = `M`.`ipv4_address````

**Comment by laf** [comment]
Thanks for the feedback but I'm not sure this is something to include to resolve specific issues like this.

**Comment by keeperofdakeys** [comment]
Yeah, I agree it's a bit too specific. The real problem is you have multiple ports with the corresponding mac addresses, and you need some kind of tie-breaking algorithm when you have multiple ports, of which two may be correct.

You have the vlan interface, as above. In this case the ip address is a tie-breaker. But this may break with VRFs, or IPs being assigned to multiple places.

You also have the physical interface (eth0). I can't seem to find any great tie-breaking strategy for my devices in the ports table besides the name. Perhaps the shortest name is a decent tie-breaker? I could see some issues with this too.

Do you have any thoughts on ways this could be fixed, or would you prefer to leave it in its semi-random state?
